src/hal/src/lib.rs

```diff
@@ -241,11 +241,21 @@ bitflags! {
         /// Allow descriptor arrays to be unsized in shaders
         const UNSIZED_DESCRIPTOR_ARRAY = 0x0800_0000_0000_0000;
         /// Mask for all the features associated with descriptor indexing.
-        const DESCRIPTOR_INDEXING_MASK = Features::SAMPLED_TEXTURE_DESCRIPTOR_INDEXING.bits | Features::STORAGE_TEXTURE_DESCRIPTOR_INDEXING.bits | Features::UNSIZED_DESCRIPTOR_ARRAY.bits;
+        const DESCRIPTOR_INDEXING_MASK = Features::SAMPLED_TEXTURE_DESCRIPTOR_INDEXING.bits | Features::STORAGE_TEXTURE_DESCRIPTOR_INDEXING.bits | Features::UNSIZED_DESCRIPTOR_ARRAY.bits | Features::UNIFORM_BUFFER_DESCRIPTOR_INDEXING.bits | Features::STORAGE_BUFFER_DESCRIPTOR_INDEXING.bits;
 
         /// Enable draw_indirect_count and draw_indexed_indirect_count
         const DRAW_INDIRECT_COUNT = 0x1000_0000_0000_0000;
 
+        /// Support for arrays of uniform buffer descriptors
+        const UNIFORM_BUFFER_DESCRIPTOR_ARRAY = 0x2000_0000_0000_0000;
```

Comment:
Not sure if its worth merging `UNIFORM_BUFFER_DESCRIPTOR_ARRAY` and `STORAGE_BUFFER_DESCRIPTOR_ARRAY` array into one feature. Is there hardware that only supports one and not the other?

Reply:
Right. Let's merge them into one.